v1/contracts/Voting.sol

```diff
@@ -33,9 +99,109 @@ contract Voting {
      * committer can reveal their vote.
      */
     function revealVote(bytes32 identifier, uint time, int price, int salt) external {
-        VoteInstance storage voteInstance = requests[identifier][time];
+        VoteInstance storage voteInstance = priceResolutions[identifier][time].votes[currentRoundNumber];
         require(keccak256(abi.encode(price, salt)) == voteInstance.committedHashes[msg.sender],
                 "Committed hash doesn't match revealed price and salt");
         delete voteInstance.committedHashes[msg.sender];
     }
+
+    /**
+     * @notice Enqueues a request (if a request isn't already present) for the given `identifier`, `time` pair.
+     * @dev Returns the time at which the user should expect the price to be resolved.
+     */
+    function requestPrice(bytes32 identifier, uint time) external returns (uint expectedTime) {
+        uint priceResolutionRound = priceResolutions[identifier][time].lastVotingRound;
+        uint secondsInWeek = 60*60*24*7;
+
+        if (priceResolutionRound == 0) {
+            // Price has never been requested.
+            // Add price request to the next round.
+            _addPriceRequestToRound(_calcVotingRound().add(1), PriceRequest(identifier, time));
+
+            // TODO: replace this with the end of the next round once phasing has been implemented.
+            return getCurrentTime().add(secondsInWeek.mul(2));
+        } else if (priceResolutionRound == currentRoundNumber) {
+            // Price is currently being resolved.
+            // TODO: replace this with the end of the current round once phasing has been implemented.
+            return getCurrentTime().add(secondsInWeek);
+        } else {
+            // Price has been resolved.
+            return 0;
+        }
+
+    }
+
+    /**
+     * @notice Gets the price for `identifier` and `time` if it has already been requested and resolved.
+     */
+    function getPrice(bytes32 identifier, uint time) external view returns (int price) {
+        PriceResolution storage priceResolution = priceResolutions[identifier][time];
```

Comment:
Yeah, this is something I've had an internal debate about. It feels like the way we're doing it now is super wasteful, but since keys aren't stored in mappings, and the mappings themselves don't take any space, it's really just an extra hash on every lookup. If we manually hashed them the way you're suggesting, we would get a savings of about 30 gas per transaction that does this lookup, but it would also require the syntax to be a bit more complex since we'd have to manually compute the hash everywhere rather than using traditional map notation. If we did it that way, I'd say we should probably just make a private method to handle it. Not super opinionated - what do you think?

Reply:
Kinda leaning towards just computing the hash ourselves.

Reply:
If its something we have to use everywhere doing it ourselves in a private method seems like the cleanest option

Reply:
Because computing a hash requires concatenating identifier and time into a `bytes` array using `abi.encode`, I figured we could just use the `bytes` as the key for the map directly so we have to compute one less hash. Anyway, I made this change, so I think this is done :)